services/parent/pom.xml

```diff
@@ -264,6 +270,144 @@
           <artifactId>quarkus-micrometer-registry-prometheus</artifactId>
         </dependency>
       </dependencies>
+    </profile>    
+    
+    <profile>
+      <id>read-ports</id>         
+      <activation>
+        <property>
+          <name>read-ports</name>
+        </property>
+      </activation>
+      <properties>
+        <docker.dir>${project.basedir}/../../tests/target/docker</docker.dir>
```

Comment:
I like the idea of reading the ports from the properties files instead of using fixed ports. Maybe we could use a path within the user's $HOME directory to keep the properties files, thus eliminating the need to use relative paths across modules? We could then also define this path in the top level POM so that we only need to define it in a single place ...

Reply:
You mean to move tests/target/docker folder outside tests folder (as certificate files are), e.g. <root>/dockerports? And add it in .gitignore. This probably could be done. 
However, regarding fixed ports, I tried to support random ports as much as possible (via profiles) but there are some problems:
- if you start local service in normal way using `mvn quarkus:dev` (the same I think does IDE) it can't launch a profile to read properties (at least I haven't found a way). Probably I could put reading in default plugins but this would affect not only live service use cases but also build. I tried to affect the existing scenarios as little as possible.
- the main disadvantage of random ports is that if you want to do something with started system, e.g. launch mongodb viewer, kafka topics reader, to connect device simulator - you should always check current random port and to configure tools. With fixed ports it is much easier.

Reply:
I see your point. Here's another idea:
instead of using the `.env` files (which do not take advantage of maven's resource filtering), we could employ Quarkus' profile mechanism and simply add the [(predefined) `dev` profile](https://quarkus.io/guides/config-reference#profiles) properties to the *application.properties* files that we already have in the adapters and services. These files can be resource filtered and thus could take advantage of centrally managed fixed port definition (e.g. in the top level POM file). The `dev` profile is automatically enabled when running a component using the *quarkus:dev* goal.

Reply:
I thought for that option but this means that _application.properties_ file would contain dev profile in images that would be built for production use. So, I decided, based on @calohmn proposal to use _.env_ files which won't affect images.

Reply:
I do not quite see how that would be a problem. If we want to make sure that we do not include these in distributed images, we might create separate *env-application.properties* files and exclude them from packaging ...

Reply:
If this is not a problem I could go for _application_dev.properties_ files which could hold dev properties.
Eventually I would try to remove them from docker image. However the last time I haven't found a way to discard files from there - so it might not succeed.

Reply:
FMPOV this would be fine. Anybody who wants to run the component using quarkus:dev with different property values can use a separate `.env` file or simply overwrite the values in the properties file ...
@calohmn WDYT?

Reply:
I'm fine with using `application_dev.properties` files and prefer them over checked-in `.env` files.
(My original rough idea was to let the `.env` files be generated when starting the containers in the `tests` module, filling the files with the variable values used there. But I guess defining the different copy-target directories and defining which properties to use specifically for each file could be too cumbersome compared with just using checked-in `application_dev.properties` files in the individual modules.)

Reply:
Then let's go this way :-)